iphone/TitaniumKit/TitaniumKit/Sources/API/TiApp.m

```diff
@@ -1197,6 +1211,111 @@ - (KrollBridge *)krollBridge
 
 #pragma mark Background Tasks
 
+- (void)registerBackgroundTasks
+{
+  NSArray *identifiers = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"BGTaskSchedulerPermittedIdentifiers"];
+
+  for (NSString *identifier in identifiers) {
+    if (registeredBackgroundTasks == nil) {
+      registeredBackgroundTasks = [[NSMutableArray alloc] init];
+    }
+    [[BGTaskScheduler sharedScheduler] registerForTaskWithIdentifier:identifier
+                                                          usingQueue:nil
+                                                       launchHandler:^(__kindof BGTask *_Nonnull task) {
+                                                         [registeredBackgroundTasks addObject:task];
+                                                         [self handleBGTask:task];
+                                                       }];
+  }
+}
+
+- (void)handleBGTask:(BGTask *)task
+{
+  NSString *notificationName = kTiBackgroundProcessNotification;
+  if ([task isKindOfClass:[BGAppRefreshTask class]]) {
+    // Fo refresh task submit it again
+    [self submitTaskForIdentifier:task.identifier];
+    notificationName = kTiBackgroundFetchNotification;
+  }
+  NSString *key = [NSString stringWithFormat:@"BgTask-%@", task.identifier];
+
+  [self tryToPostBackgroundModeNotification:[NSMutableDictionary dictionaryWithObjectsAndKeys:key, @"handlerId", nil]
+                       withNotificationName:notificationName];
+
+  task.expirationHandler = ^{
+    if ([task isKindOfClass:[BGProcessingTask class]]) {
+      // Fo processing task, if it is not completed in time then only submit it again.
+      [self submitTaskForIdentifier:task.identifier];
+    }
+    [task setTaskCompletedWithSuccess:false];
+    [registeredBackgroundTasks removeObject:task];
+  };
+}
+
+- (void)submitTaskForIdentifier:(NSString *)identifier
+{
+  NSDictionary *backgroundTask = [self backgroundTaskForIdentifier:identifier];
+  if (backgroundTask) {
+    [self submitBackgroundTask:backgroundTask];
+  }
+}
+
+- (void)submitBackgroundTask:(NSDictionary *)bgTask
+{
+  BGTaskRequest *taskRequest;
+  if ([bgTask[@"type"] isEqualToString:@"process"]) {
+    taskRequest = [[[BGProcessingTaskRequest alloc] initWithIdentifier:bgTask[@"identifier"]] autorelease];
+    ((BGProcessingTaskRequest *)taskRequest).requiresNetworkConnectivity = [TiUtils boolValue:bgTask[@"powerConnect"] def:NO];
```

Comment:
Looks like you have this and the next line swapped as to which dictionary key/value pair it should use.
```suggestion
    ((BGProcessingTaskRequest *)taskRequest).requiresNetworkConnectivity = [TiUtils boolValue:bgTask[@"networkConnect"] def:NO];
```